### ISSUE TITLE
Request - geog lookup tool - data request

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#8785]
Help us understand your request (check below):

- [x] other

**Describe what you're trying to do** 

Actual data would help with https://github.com/ArctosDB/dev/issues/210. If anyone has any "wild" geography data that needs Actos-ified, please pass it along!

@ekrimmel @acdoll

**Comment by ekrimmel** [comment]
Here is some wild geography, straight from an incoming collection. There are a handful of marine bits tucked into _locality_, if you care to use that (otherwise you can ignore the whole column).

[FHSM-Fish-2025-04-16.csv](https://github.com/user-attachments/files/24534782/FHSM-Fish-2025-04-16.csv)

**Comment by dustymc** [comment]
Reopening, there's apparently interest/need!

>handful of marine bits

... but IDK how I'm going to handle that; the lookup tool was built when there was (meant to be) about one way to say "there." Our current model supports lots eg:

* Some Ocean, Whatever Sea (which one might use because those things are relatively well-defined)
* Some Ocean, Whatever Sea, Some Bay (which one might avoid because bays tend to have fuzzy edges)
* Some Ocean, Whatever Sea, That Country (which one might use to indicate something political)
* Some Ocean, Whatever Sea, Some Bay, That Country (all of the "edge" components can be used or not in combination with lots of other stuff)
* Some Ocean, Whatever Sea, That Country, Statey McStateface (permits?!?)
* Some Ocean, Whatever Sea, That Country, Statey McStateface, Nearby County (IDK why but we're making these)

I suppose the tool would need to return all of those if they seem relevant?? Would that be useful? 

Or possibly the tool would only "recommend" (because it'll be seen that way...) either (both??) dirt (and/?)or water, and avoid the combos?? (That doesn't seem quite right...)

For these data, the tool should be able to figure out how to translate "USA,"  confirm the spelling and placement of the state, and maybe find that typo or suggest the format of county - maybe that's enough and we can deal with anything more complex when there's a tangible use case for that??

What would be useful here? HELP!

**Comment by dustymc** [comment]
Ping @mkoo briefly discussed today, still not clear to me what function this tool could perform.

**Comment by ekrimmel** [comment]
From a data migration perspective, a simple and helpful function would be something akin to the Taxon Name Checker. I think this is what the "Higher Geog Lookup" tool used to do? I usually just want to know whether the _locality_higher_geog_ value I have in a bulkloader file is going to match an existing higher geography record. Beyond simply telling me that "United States, Florida, St. Johns County" does not exist, maybe a tool could say that "United States, Florida, **Saint** Johns County" does. But I think it would be helpful even if the tool just looked up values and said whether or not a match was found. Obviously these errors get caught and can be fixed in the bulkloader, but it's often nice to be able to fix them prior, especially when they affect many records.

Not directly related, but in looking for a quick example I learned we have "Atlantic Ocean, United States, Florida, **St.** Johns County" instead of "Atlantic Ocean, United States, Florida, Saint Johns County."

**Comment by dustymc** [comment]
>used to do

It used to do lots of stuff that probably can't make sense with arbitrary geography, which is what The Community has decided to go with.

>United States, Florida, St. Johns County" does not exist

This is definitely technically plausible.

>"United States, Florida, Saint Johns County" does

This is kinda-maybe plausible, but I'm not sure it's useful - eg maybe the county not doing what I'd have done was done for a reason, it can't really be "wrong" in this model because the data are very explicitly curatorial. (I still hope someone wants to make it less cryptic.)

>Florida

As above that is expected, we're not following any Standards so anything goes, but ugh anyway.

https://arctos.database.museum/place.cfm?sch=geog&state_prov=florida&county=john

**Comment by ekrimmel** [comment]
> This is definitely technically plausible.

So yea, IMO even just this would be great!

**Comment by Jegelewicz** [comment]
Yes - just telling me that my HG is in the table would be useful - I can manually work with whatever is left.

**Comment by acdoll** [comment]
Yay! It looks like this is up and functional, yes? Can I suggest we edit the STATUS returns to be something more obvious? maybe:
"nomatch" = "no match"
"trim ilike match" = "exact match"
"trim stripgeogranks ilike match" = "edited match" (not totally sure about this one)

Thanks for working on this - it will be REALLY helpful!!

**Comment by dustymc** [comment]
>up and functional, yes

Yup.

>obvious

Sure - but eg "trim ilike match" != "exact match" (eg your input won't necessarily make a bulkloader happy, but it's not TOO manipulated either). 

Possibly just "here's some stuff that might work" would be more clear that trying to explain the details (and it's designed to grow and change as necessary, keeping up with that might be a lot of work)?

**Comment by Jegelewicz** [comment]
Could you just supply

no match
exact match
fuzzy match

?
